tactis/model/decoder.py

```diff
@@ -130,35 +151,65 @@ def loss(self, encoded: torch.Tensor, mask: torch.BoolTensor, true_value: torch.
         loss: torch.Tensor [batch]
             The loss function, equal to the negative log likelihood of the distribution.
         """
-        encoded = _merge_series_time_dims(encoded)
+        B, S, T, E = flow_encoded.shape
+        flow_encoded = _merge_series_time_dims(flow_encoded)
+        if not self.skip_copula:
+            copula_encoded = _merge_series_time_dims(copula_encoded)
+
         mask = _merge_series_time_dims(mask)
         true_value = _merge_series_time_dims(true_value)
 
-        # Assume that the mask is constant inside the batch
-        mask = mask[0, :]
+        # Assume that the mask is constant inside the batch (every sample has the same hist-pred split)
+        mask = mask[0, :]  # (series * time steps)
 
-        hist_encoded = encoded[:, mask, :]
-        pred_encoded = encoded[:, ~mask, :]
-        hist_true_x = true_value[:, mask]
+        hist_encoded_flow = flow_encoded[:, mask, :]  # [batch, series*(num_hist_timesteps), embedding]
+        if not self.skip_copula:
+            hist_encoded_copula = copula_encoded[:, mask, :]  # [batch, series*(num_hist_timesteps), embedding]
+        pred_encoded_flow = flow_encoded[:, ~mask, :]
+        if not self.skip_copula:
+            pred_encoded_copula = copula_encoded[:, ~mask, :]
+        history_factor = hist_encoded_flow.shape[1] / pred_encoded_flow.shape[1]
+
+        hist_true_x = true_value[:, mask]  # [batch, series*(num_hist_timesteps)]
         pred_true_x = true_value[:, ~mask]
 
+        if type(T) == torch.Tensor:
+            T = T.item()
```

Comment:
This piece of code is weird. T should always be a scalar due to how you initialized it using `flow_encoded.shape`.

Reply:
Wrote those a while back since some error was there. 
Removed it now. Will debug if it ever comes up again.
Resolved.